### PR TITLE
feat!: switch to dockerized github action

### PR DESCRIPTION
**Pull request by derberg**
**Description**

- move away from pushing `node_modules` and `dist` to repo
- add release to docker workflow
- update instructions to use action from docker - I want to release an new major because of it
- Dockerfile
- all the necessary cleanup that adds noise to the PR, I'm aware but tried to have separate commits. Also cleanup is mainly file deletion, so it looks still pretty clean

---

dockerized action tried [here](https://github.com/derberg/test-experiment/blob/master/.github/workflows/main.yml#L23) with [success](https://github.com/derberg/test-experiment/actions/runs/1491061116) because [this](https://github.com/derberg/test-experiment/blob/master/docs/api/my-asyncapi.yml#L3) new title is now [here](https://derberg.github.io/test-experiment/)

### CODE REVIEW

File: .github/workflows/release-docker.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+
+  release:
+    name: Docker build and push
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: master
```
Comment (magicmatatjahu) [inline .github/workflows/release-docker.yml:16]: Is it needed?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM node:16 as builder
+
+COPY ./ /app
+WORKDIR /app
+
+RUN npm install && npm run package
+
+FROM node:16-alpine as runner
```
Comment (magicmatatjahu) [inline Dockerfile:8]:
```suggestion
FROM node:16-alpine
```

You don't need to make alias.

**Review by magicmatatjahu — APPROVED** [review]
LGTM, but I see two (not needed) improvements.

**Review by magicmatatjahu — APPROVED** [review]

File: .github/workflows/release-docker.yml (hunk 1 above)
Comment (derberg) [reply .github/workflows/release-docker.yml:16]: yes, the workflow is triggered not on push but on release event, so it doesn't know from the context what branch to checkout unfortunately 😞

Comment (magicmatatjahu) [reply .github/workflows/release-docker.yml:16]: Ah yes 🤦🏼  Forgot about that, thanks!

**Pull request merged**